src/main/scala/org/broadinstitute/dsde/workbench/sam/directory/LdapDirectoryDAO.scala

```diff
@@ -329,4 +323,17 @@ class LdapDirectoryDAO(protected val ldapConnectionPool: LDAPConnectionPool, pro
     ldapConnectionPool.modify(petDn(petServiceAccount.id), modifications.asJava)
     petServiceAccount
   }
+
+  override def loadSubjectFromGoogleSubjectId(googleSubjectId: GoogleSubjectId): Future[Option[WorkbenchSubject]] =
+    for{
+      entries <- Future{ldapConnectionPool.search(peopleOu, SearchScope.SUB, Filter.createEqualityFilter(Attr.googleSubjectId, googleSubjectId.value)).getSearchEntries.asScala}
+      res <- entries match {
+        case Seq() => Future.successful(None)
+        case Seq(subject) => Future.successful(Try(dnToSubject(subject.getDN)).toOption)
+        case subjects => Future.failed(new WorkbenchException(s"Database error: googleSubjectId $googleSubjectId refers to too many subjects: ${subjects.map(_.getDN)}"))
+      }
+    } yield res
+
+  override def setGoogleSubjectId(userId: WorkbenchUserId, googleSubjectId: GoogleSubjectId): Future[Unit] =
+    Future{ldapConnectionPool.modify(userDn(userId), new Modification(ModificationType.ADD, Attr.googleSubjectId, googleSubjectId.value))}
```

Comment:
It's just `add`. If a record already have the attribute, it'll return a failed `Future`.

I think @dvoet suggested `set` since it's used in other code with similar semantics.

Reply:
hmmm...OK, if there's a convention, then I won't try to change it, but I still think it's confusing/misleading.